src/client.ts

```diff
@@ -1,124 +1,100 @@
-      },
-      {
-        id: 'acme-user-2',
-        name: 'User Two',
-      },
-    ];
+  private computeHash(data, key) {
+    const hmac = createHmac(HASH_ALGORITHM, key);
+    hmac.update(data);
+    return hmac.digest();
+  }
 
-    for (const user of users) {
-      await iteratee(user);
-    }
+  private calculateDataSignature(apiKey, nonceBytes, dateStamp, data) {
+    const kNonce = this.computeHash(nonceBytes, Buffer.from(apiKey, 'hex'));
+    const kDate = this.computeHash(dateStamp, kNonce);
+    const kSignature = this.computeHash(REQUEST_VERSION, kDate);
+    return this.computeHash(data, kSignature);
   }
 
-  /**
-   * Iterates each group resource in the provider.
-   *
-   * @param iteratee receives each resource to produce entities/relationships
-   */
-  public async iterateGroups(
-    iteratee: ResourceIteratee<AcmeGroup>,
-  ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
+  private calculateAuthorizationHeader(urlString, httpMethod) {
+    const { apiId, apiSecret } = this.config;
+    const url = new URL(urlString);
+    const hostName = url.hostname;
+    const veraURL = url.pathname + url.search;
+    const data = `id=${apiId}&host=${hostName}&url=${veraURL}&method=${httpMethod}`;
+    const dateStamp = this.currentDateStamp();
+    const nonceBytes = this.newNonce(NONCE_SIZE);
+    const dataSignature = this.calculateDataSignature(
+      apiSecret,
+      nonceBytes,
+      dateStamp,
+      data,
+    );
+    const authorizationParam = `id=${apiId},ts=${dateStamp},nonce=${nonceBytes.toString(
+      'hex',
+    )},sig=${dataSignature.toString('hex')}`;
+    return `${AUTH_SCHEME} ${authorizationParam}`;
+  }
 
-    const groups: AcmeGroup[] = [
-      {
-        id: 'acme-group-1',
-        name: 'Group One',
-        users: [
-          {
-            id: 'acme-user-1',
-          },
-        ],
+  // call v1 applications route and v2 findings route to ensure adequate permissions are enabled
+  public async verifyAuthentication(): Promise<void> {
+    const applicationsEndpoint = BASE_URI_V1 + 'applications';
+    const applicationsRequest = got.get(applicationsEndpoint, {
+      headers: {
+        Authorization: this.calculateAuthorizationHeader(
+          applicationsEndpoint,
+          'GET',
+        ),
       },
-    ];
-
-    for (const group of groups) {
-      await iteratee(group);
+      retry: gotRetryOptions,
+    });
+    try {
+      await applicationsRequest;
+    } catch (err) {
+      throw new IntegrationProviderAuthenticationError({
+        cause: err,
+        endpoint: applicationsEndpoint,
+        status: err.status,
```

Comment:
I think for `got` status will be `err.respsonse?.statusCode` and statusText will be `err.response?.statusMessage`.
[Issue From Rumble](https://github.com/JupiterOne/graph-rumble/pull/2#discussion_r799768259)
[Got Error Docs](https://github.com/sindresorhus/got/blob/HEAD/documentation/8-errors.md)

Reply:
good catch!